apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue

```diff
@@ -1,66 +1,94 @@
-import {addDays, isSameDay, isToday} from 'date-fns';
-import {CLOSED, OPENING_HOURS, SHOW_MORE_HOURS} from '@myparcel-do/shared';
+import {computed, onDeactivated, ref, toRefs} from 'vue';
+import {
+  CLOSED,
+  OPENING_HOURS,
+  SHOW_MORE_HOURS,
+  DAY_MONDAY,
+  DAY_TUESDAY,
+  DAY_WEDNESDAY,
+  DAY_THURSDAY,
+  DAY_FRIDAY,
+  DAY_SATURDAY,
+  DAY_SUNDAY,
+} from '@myparcel-do/shared';
 import {type StartEndDate} from '@myparcel/sdk';
-import {createNextDate, createUtcDate} from '../../../../utils';
+import {createNextDate} from '../../../../utils';
 import {SHOWN_OPENING_HOURS} from '../../../../data';
-import {useDateFormat, useLanguage, usePickupLocation, useTimeRange} from '../../../../composables';
+import {useDateFormat, useLanguage, usePickupLocation} from '../../../../composables';
 import {DoButton} from '../../../../components';
 
-const props = defineProps<{locationCode: string; expanded?: boolean}>();
+const props = defineProps<{ locationCode: string; expanded?: boolean }>();
 const propRefs = toRefs(props);
 
-const {translate} = useLanguage();
-
-const {pickupLocation} = usePickupLocation(propRefs.locationCode);
+const { translate } = useLanguage();
+const { pickupLocation } = usePickupLocation(propRefs.locationCode);
 
 const mutableShowAll = ref(false);
 const resolvedShowAll = computed(() => props.expanded || mutableShowAll.value);
 
+const weekDaysMap = {
+  [DAY_MONDAY]: 'Monday',
+  [DAY_TUESDAY]: 'Tuesday',
+  [DAY_WEDNESDAY]: 'Wednesday',
+  [DAY_THURSDAY]: 'Thursday',
+  [DAY_FRIDAY]: 'Friday',
+  [DAY_SATURDAY]: 'Saturday',
+  [DAY_SUNDAY]: 'Sunday',
+};
+
 const openingHours = computed(() => {
-  return (pickupLocation.value?.openingHours ?? [])
-    .map(({hours, weekday}) => {
-      const date = createNextDate(weekday);
-      const formattedDay = useDateFormat(date);
-
-      const isTodayOrTomorrow = isToday(date) || isSameDay(addDays(createUtcDate(), 1), date);
-
-      const time: StartEndDate = hours?.[0];
-      const timeString = time ? useTimeRange(time.start.date, time.end.date).value : translate(CLOSED);
-
-      return {
-        date,
-        weekday: capitalize(isTodayOrTomorrow ? formattedDay.relative.value : formattedDay.weekday.value),
-        timeString,
-      };
-    })
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
+  const allWeekdays = [DAY_SUNDAY, DAY_MONDAY, DAY_TUESDAY, DAY_WEDNESDAY, DAY_THURSDAY, DAY_FRIDAY, DAY_SATURDAY];
+
+  const hoursArray = allWeekdays.map((weekday) => {
+    const openingHour = (pickupLocation.value?.openingHours ?? []).find(({ weekday: wd }) => wd === weekday);
+    const date = createNextDate(weekday);
+    const formattedDay = useDateFormat(date);
+
+    const time: StartEndDate | undefined = openingHour && Array.isArray(openingHour.hours) && openingHour.hours.length > 0 ? openingHour.hours[0] : undefined;
+
+    let timerange = 'CLOSED';
+    if (time) {
+      const start = time.start.date.substring(11, 16);
```

Comment:
Is there a way to get this data without using `substring()`? I am worried that this might not work in the future if the formatting of date changes.

Reply:
I agree this can be clearer (no magic numbers), but the string is a fixed ISO string that will likely not change over the lifetime of the application.